src/blockchain/block_tree.hpp

```diff
@@ -150,6 +152,26 @@ namespace lean::blockchain {
      * @return hash of the block
      */
     [[nodiscard]] virtual BlockIndex lastFinalized() const = 0;
+
+    /**
+     * Get message for "/leanconsensus/req/status/1/ssz_snappy" protocol.
+     * Returns hash and slot for finalized and best blocks.
+     */
+    virtual StatusMessage getStatusMessage() const = 0;
```

Comment:
BlockTree should not know about network-things. This code violates S in SOLID.

Reply:
This function reuses existing type, and returns two things in one call